specification/ai/Azure.AI.Projects/servicepatterns.tsp

```diff
@@ -0,0 +1,121 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+
+namespace Azure.AI.Projects.ServicePatterns {
+  @Rest.autoRoute
+  interface VersionedOperations<
+    TEntityType extends Reflection.Model,
+    TListLatestParameters extends Reflection.Model = {},
+    TListVersionsParameters extends Reflection.Model = {}
+  > {
+    #suppress "@azure-tools/typespec-azure-core/verb-conflict" "We are using the resource action as a building block."
+    @doc("List all versions of the given {name}", TEntityType)
+    @Rest.action("versions")
+    @Rest.actionSeparator("/")
+    @Http.get
+    listVersions is BuildingBlocks.CoreOps.ResourceAction<
+      TEntityType,
+      TListVersionsParameters,
+      Azure.Core.Page<TEntityType>
+    >;
+
+    @doc("List the latest version of each {name}", TEntityType)
+    @Rest.listsResource(TEntityType)
+    listLatest is Azure.Core.Foundations.ResourceList<
+      TEntityType,
+      TListLatestParameters,
+      Azure.Core.Page<TEntityType>
+    >;
+
+    @doc("Get the specific version of the {name}", TEntityType)
+    getVersion is Azure.Core.Foundations.ResourceOperation<
+      TEntityType,
+      {
+        @doc("The specific version id of the {name} to retrieve.", TEntityType)
+        @Rest.segment("versions")
+        @Http.path
+        version: string;
+      },
+      Azure.Core.Foundations.ResourceOkResponse<TEntityType>
+    >;
+
+    @doc("Delete the specific version of the {name}", TEntityType)
+    @Http.delete
+    deleteVersion is Azure.Core.Foundations.ResourceOperation<
+      TEntityType,
+      {
+        @doc("The version of the {name} to delete.", TEntityType)
+        @Rest.segment("versions")
+        @Http.path
+        version: string;
+      },
+      Http.NoContentResponse
+    >;
+
+    @doc(
+      "Create a new {name}. The version id will be generated by the service.",
+      TEntityType
+    )
+    @Rest.actionSeparator("/")
+    versions is BuildingBlocks.RepeatableCoreOps.ResourceAction<
```

Comment:
Per IM conversation, I think this should be called "create" (so we have "create" and "create_version").